src/smac_planner_hybrid.cpp

```diff
@@ -218,6 +220,43 @@ PlanResult SmacPlannerHybrid::planWithWaypoint(
   return result;
 }
 
+std::vector<geometry_msgs::PoseStamped> SmacPlannerHybrid::computeWaypoints(const geometry_msgs::PoseStamped& start, const geometry_msgs::PoseStamped& goal, float tolerance) {
+  geometry_msgs::PoseStamped waypoint_front;
+  geometry_msgs::PoseStamped waypoint_back;
+  waypoint_front.pose = Utils::getPoseAtDistanceAlongHeading(goal.pose, _search_info.goal_align_distance);
+  waypoint_back.pose = Utils::getPoseAtDistanceAlongHeading(goal.pose,  -_search_info.goal_align_distance);
+  waypoint_front.header = goal.header;
+  waypoint_back.header = goal.header;
+
+  if (Utils::isSamePose(goal.pose, waypoint_front.pose, tolerance)) {
+    return {};
+  }
+
+  std::vector<geometry_msgs::PoseStamped> waypoints;
+
+  const bool can_use_front_waypoint = !NodeHybrid::arePosesSameDiscreteState(waypoint_front.pose, start.pose, _costmap);
+  const bool can_use_back_waypoint = !NodeHybrid::arePosesSameDiscreteState(waypoint_back.pose, start.pose, _costmap);
+
```

Comment:
Q/R: can we check here if we are at any waypoint first, and return in this case?

```suggestion
  if (!can_use_front_waypoint || !can_use_back_waypoint) {
     return {};
  }
```

Then we don't need to check `can_use_front_waypoint` or `can_use_back_waypoint` again below. We can just do:

```
if (!_search_info.allow_goal_overshoot) {
    if (_search_info.isStartBehindSearchBounds()) {
      ROS_INFO_NAMED("smac_planner_hybrid", "waypoint is %f meters back of the goal pose", _search_info.goal_align_distance);
      return {waypoint_back};
    }
    else{
      return {waypoint_front};
    }
}
else {
  return {waypoint_front, waypoint_back};
}
```

Reply:
hey, I have changed it pls check